recipes/tbb/all/conanfile.py

```diff
@@ -0,0 +1,153 @@
+        if self.settings.os == "Windows" and self.options.tbbproxy and \
+           not self.options.tbbmalloc:
+            raise ConanInvalidConfiguration("tbbproxy needs tbbmaloc and shared options")
+        if self.settings.os == "Windows" and not self.options.shared:
+            raise ConanInvalidConfiguration("TBB could not be built as static lib on Windows")
+
+    @property
+    def is_msvc(self):
+        return self.settings.compiler == 'Visual Studio'
+
+    @property
+    def is_mingw(self):
+        return self.settings.os == 'Windows' and self.settings.compiler == 'gcc'
+
+    @property
+    def is_clanglc(self):
+        return self.settings.os == 'Windows' and self.settings.compiler == 'clang'
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name.lower(), self.version.upper()), self._source_subfolder)
+
+        # Get the version of the current compiler instead of gcc
+        linux_include = os.path.join(self._source_subfolder, "build", "linux.inc")
+        tools.replace_in_file(linux_include, "shell gcc", "shell $(CC)")
+        tools.replace_in_file(linux_include, "= gcc", "= $(CC)")
+
+    def get_targets(self):
+        targets = ["tbb"]
+        if self.options.tbbmalloc:
+            targets.append("tbbmalloc")
+        if self.options.tbbproxy:
+            targets.append("tbbproxy")
+        return targets
+    
+    def build(self):
+        def add_flag(name, value):
+            if name in os.environ:
+                os.environ[name] += ' ' + value
+            else:
+                os.environ[name] = value
+
+        extra = "" if self.settings.os == "Windows" or self.options.shared else "extra_inc=big_iron.inc"
+        if self.settings.arch == "x86":
+            arch = "ia32" 
+        elif self.settings.arch == "x86_64":
+            arch = "intel64"
+        elif self.settings.arch == "armv7":
+            arch = "armv7"
+        elif self.settings.arch == "armv8":
+            arch = "aarch64"
+        if self.settings.compiler in ['gcc', 'clang', 'apple-clang']:
+            if str(self.settings.compiler.libcxx) in ['libstdc++', 'libstdc++11']:
+                extra += " stdlib=libstdc++"
+            elif str(self.settings.compiler.libcxx) == 'libc++':
+                extra += " stdlib=libc++"
+            extra += " compiler=gcc" if self.settings.compiler == 'gcc' else " compiler=clang"
+
+            extra += " gcc_version={}".format(str(self.settings.compiler.version))
+        make = tools.get_env("CONAN_MAKE_PROGRAM", tools.which("make") or tools.which('mingw32-make'))
+        if not make:
+            raise ConanInvalidConfiguration("This package needs 'make' in the path to build")
+
+        with tools.chdir(self._source_subfolder):
+            # intentionally not using AutoToolsBuildEnvironment for now - it's broken for clang-cl
```

Comment:
Is there an issue for this in conan-io/conan repo?

Reply:
yes, there are multiple issues and PRs to support `clang-cl` in conan project already